moveit_ros/planning_interface/move_group_interface/include/moveit/move_group_interface/move_group_interface.h

```diff
@@ -740,30 +740,113 @@ class MoveGroupInterface
       between end effector configurations of consecutive points in the result \e trajectory. The reference frame for the
-      waypoints is that specified by setPoseReferenceFrame(). No more than \e jump_threshold
-      is allowed as change in distance in the configuration space of the robot (this is to prevent 'jumps' in IK
-     solutions).
+      waypoints is that specified by setPoseReferenceFrame(). \e jump_threshold defines a factor off the average
+      configuration space distance between waypoints that is acceptable between any two waypoints (this is to prevent
+      'jumps' in IK solutions).
       Collisions are avoided if \e avoid_collisions is set to true. If collisions cannot be avoided, the function fails.
       Return a value that is between 0.0 and 1.0 indicating the fraction of the path achieved as described by the
-     waypoints.
+      waypoints.
       Return -1.0 in case of error. */
   double computeCartesianPath(const std::vector<geometry_msgs::Pose>& waypoints, double eef_step, double jump_threshold,
+                              moveit_msgs::RobotTrajectory& trajectory, bool avoid_collisions = true,
+                              moveit_msgs::MoveItErrorCodes* error_code = nullptr)
+  {
+    return computeCartesianPath(waypoints, eef_step, jump_threshold, 0, 0, trajectory, avoid_collisions, error_code);
+  }
+
+  /** \brief Compute a Cartesian path that follows specified waypoints with a step size of at most \e eef_step meters
+      between end effector configurations of consecutive points in the result \e trajectory. The reference frame for the
+      waypoints is that specified by setPoseReferenceFrame(). \e jump_threshold defines a factor off the average
+      configuration space distance between waypoints that is acceptable between any two waypoints (this is to prevent
+      'jumps' in IK solutions).
+      Kinematic constraints for the path given by \e path_constraints will be met for every point along the trajectory,
+      if they are not met, a partial solution will be returned.
+      Constraints are checked (collision and kinematic) if \e avoid_collisions is set to true. If constraints cannot be
+      met, the function fails.
+      Return a value that is between 0.0 and 1.0 indicating the fraction of the path achieved as described by the
+      waypoints.
+      Return -1.0 in case of error. */
+  double computeCartesianPath(const std::vector<geometry_msgs::Pose>& waypoints, double eef_step, double jump_threshold,
+                              moveit_msgs::RobotTrajectory& trajectory,
+                              const moveit_msgs::Constraints& path_constraints, bool avoid_collisions = true,
+                              moveit_msgs::MoveItErrorCodes* error_code = nullptr)
+  {
+    return computeCartesianPath(waypoints, eef_step, jump_threshold, 0, 0, trajectory, path_constraints,
+                                avoid_collisions, error_code);
+  }
```

Comment:
I don't like this explosion into various variants of the function at all. What about, just providing a single variant accepting the `JumpThreshold` class? The old variants should be deprecated and removed later.

Reply:
> I don't like this explosion into various variants of the function at all. What about, just providing a single variant accepting the `JumpThreshold` class? The old variants should be deprecated and removed later.

I think the consensus was that this would be a better solution.  IF people don't mind, I think this PR should be closed and if/when I or somebody else gets around to fixing this in a cleaner way, a new PR can be opened.